src/graphics/transformation.rs

```diff
@@ -33,35 +34,67 @@ impl Transformation {
     pub fn translate(translation: Vector) -> Transformation {
-        Transformation(nalgebra::Matrix4::new_translation(
-            &nalgebra::Vector3::new(translation.x, translation.y, 0.0),
-        ))
+        Transformation(Matrix4::new_translation(&Vector3::new(
+            translation.x,
+            translation.y,
+            0.0,
+        )))
     }
 
     /// Creates a uniform scale transformation.
     ///
     /// You can use this to zoom your camera, for example.
     pub fn scale(scale: f32) -> Transformation {
-        Transformation(nalgebra::Matrix4::new_scaling(scale))
+        Transformation(Matrix4::new_scaling(scale))
     }
 
     /// Creates a non-uniform scale transformation.
     ///
     /// It allows you to scale each axis independently. You should rarely need
     /// this.
-    pub fn nonuniform_scale(x: f32, y: f32) -> Transformation {
-        Transformation(nalgebra::Matrix4::new_nonuniform_scaling(
-            &nalgebra::Vector3::new(x, y, 1.0),
-        ))
+    pub fn nonuniform_scale(scale: Vector) -> Transformation {
+        Transformation(Matrix4::new_nonuniform_scaling(&Vector3::new(
+            scale.x, scale.y, 1.0,
+        )))
     }
 
     /// Creates a rotation transformation (in radians).
     ///
     /// You can use this to rotate your camera, for example.
     pub fn rotate(rotation: f32) -> Transformation {
-        Transformation(nalgebra::Matrix4::new_rotation(nalgebra::Vector3::new(
-            0.0, 0.0, rotation,
-        )))
+        Transformation(Matrix4::new_rotation(Vector3::new(0.0, 0.0, rotation)))
+    }
+
+    /// Transforms the given point by this transformation.
+    pub fn transform_point(self, point: Point) -> Point {
+        let point = self.0.transform_point(&Point3::new(point.x, point.y, 0.0));
+        Point::new(point.x, point.y)
+    }
+
+    /// Transforms the given vector by this transformation.
+    pub fn transform_vector(self, vector: Vector) -> Vector {
+        let vector = self
+            .0
+            .transform_vector(&Vector3::new(vector.x, vector.y, 0.0));
+        Vector::new(vector.x, vector.y)
+    }
+
+    /// Transforms the given point by the inverse of this transformation.
+    pub fn inverse_transform_point(self, point: Point) -> Point {
+        let point = self.0
+            .try_inverse()
+            .expect("Transformation matrix should only contain invertible operations")
+            .transform_point(&Point3::new(point.x, point.y, 0.0));
+        Point::new(point.x, point.y)
+    }
+
+    /// Transforms the given vector by the inverse of this transformation.
+    pub fn inverse_transform_vector(self, vector: Vector) -> Vector {
+        let vector = self.0
+            .try_inverse()
+            .expect("Transformation matrix should only contain invertible operations")
+            .transform_vector(&Vector3::new(vector.x, vector.y, 0.0));
+        Vector::new(vector.x, vector.y)
```

Comment:
These methods feel out of scope, for now at least. I think users can do this by themselves by turning the `Transformation` into a `nalgebra::Matrix4` with the brand new trait implementations.

We shouldn't try to reimplement/wrap a math library.

Reply:
As said on discord, I disagree, since it:

1. Requires knowledge about the underlying math library
2. Requires adding and removing a third dimension (at least until Transformation uses a 3x3 matrix internally)